spec/rest_in_peace/template_sanitizer_spec.rb

```diff
@@ -120,12 +120,20 @@
 
       context 'param which need to be encoded' do
         let(:params) { { id: 'this param need to be encoded' } }
-        let(:encoded_param) { 'this+param+need+to+be+encoded' }
+        let(:encoded_param) { 'this%20param%20need%20to%20be%20encoded' }
```

Comment:
Ouch, my bad!

But then I must say, after some research, I'm hesitant to approve this change. I'd argue as follows:

The basics of the `URL` are defined in [rfc1738](https://tools.ietf.org/html/rfc1738#section-2.1). In it, it is only defined that "+" is allowed as special character, but not it's meaning.

None of the applicable errata of rfc1738, namely rfc1808, rfc2396, rfc3986, define the meaning of the "+" character any further.

It is the "[URL Standard](https://url.spec.whatwg.org/#urlencoded-serializing)" by the WHATWG then, that finally sheds some light. It's section 5.2 defines, how a parameter shall be encoded to conform `application/x-www-form-urlencoded`. And it says:

    The application/x-www-form-urlencoded byte serializer
    takes a byte sequence `input` and then runs these steps:
    
      1. Let `output` be the empty string.
      2. For each byte in `input`, depending on byte:
          0x20 (SP)         Append U+002B (+) to `output`.
          [...]
      3. Return `output`.

In other terms, the first step of a standards-conform serialiser has to be replacing any ` ` it encounters by `+`. The current implementation seems to do this correct, whereas your patch would change the behaviour to something else.

This can now be seen as bikeshedding, probably rightful so, especially as all standards-conform URL parser will accept the `%20` encoding perfectly well. But so should they accept the `+` encoding. I'm interested therefore: What problem you are trying to solve with the proposed change? Would you mind to elaborate?